src/main/kotlin/no/nav/syfo/behandlerdialog/MeldingFraBehandlerService.kt

```diff
@@ -38,13 +38,11 @@ class MeldingFraBehandlerService(
         if (melding.parentRef != null) {
             database.connection.use { connection ->
                 val existingOppgave = connection
-                    .getPersonOppgaveByReferanseUuid(melding.parentRef)
-                    ?.toPersonOppgave()
+                    .getPersonOppgaverByReferanseUuid(melding.parentRef)
+                    .map { it.toPersonOppgave() }
+                    .firstOrNull { it.type == PersonOppgaveType.BEHANDLERDIALOG_MELDING_UBESVART && it.isUBehandlet() }
```

Comment:
Går det an å sjekke typen før man mapper? I så fall kunne man sluppet å mappe hele listen 🤔

Reply:
Ja, antakelig mulig, men tror at firstOrNull-testen vil bli litt mer innfløkt hvis vi skal gjøre det. Siden det bare i svært sjeldne tilfeller er mer enn en personoppgave for en referanseUUID så synes jeg det er best at koden er slik den er.